backend/coins/eth/account.go

```diff
@@ -213,6 +214,10 @@ func (account *Account) poll() {
 			} else {
 				account.SetOffline(false)
 			}
+			if decRequestCounter != nil {
+				decRequestCounter()
```

Comment:
Not sure how everything works together here but it seems the account assume to be synchronized ater the first loop iteration unconditionally, even if `account.update()` was unsuccessful. Is that intended?

Also, I just happened to notice there seem to be 2 separate ways in which an account is assumed to be syncronized:

1. In Synchronizer.WaitSynchronized, which waits on a channel until it's closed and it's what's used here.

```go
// WaitSynchronized blocks until all pending synchronization tasks are finished.
func (synchronizer *Synchronizer) WaitSynchronized() {
	synchronizer.log.WithFields(logrus.Fields{"requestCounter": synchronizer.requestsCounter}).
		Debug("wait synchronized")
	if func() int32 {
		defer synchronizer.waitLock.RLock()()
		return synchronizer.requestsCounter
	}() == 0 {
		return
	}
	<-synchronizer.wait
}
```

2. BaseAccount sets `synced` field to `true` and emits an event:

```go
// NewBaseAccount creates a new Account instance.
func NewBaseAccount(config *AccountConfig, coin coin.Coin, log *logrus.Entry) *BaseAccount {
	account := &BaseAccount{
		config: config,
		coin:   coin,
	}
	account.Synchronizer = synchronizer.NewSynchronizer(
		func() { config.OnEvent(EventSyncStarted) },
		func() {
			if !account.synced {
				account.synced = true
				config.OnEvent(EventStatusChanged)
			}
			config.OnEvent(EventSyncDone)
		},
		log,
	)
	return account
}
```

Seems like they both serve the same purpose? If so, maybe they could be combined into one in the future.

Reply:
WaitSynchronized() waits until the current round of updating is finished. In BTC: when a addresses are re-synced cause a new tx touches them, in eth: in another poll() iteration. Balance() should wait until that is done to return to ensure a consistent state.

SycDone only refers to the "first" such sync. before that, there is no meaningful data at all.

Reply:
> Not sure how everything works together here but it seems the account assume to be synchronized ater the first loop iteration unconditionally, even if account.update() was unsuccessful. Is that intended?

Yeah. Currently, failure states are handled via the Offline() and FatalError() (the Offline() one we actually don't handle yet in the portfolio, but should).

Reply:
I have to say it isn't clear at all to someone who doesn't know the code by heart. Still, not sure I understand:

> in eth: in another poll() iteration

so, that's 2 iterations in total? if so, why is `if decRequestCounter != nil { decRequestCounter() }` executed in the 1st iteration?

Reply:
> I have to say it isn't clear at all to someone who doesn't know the code by heart.

Agree, I'd like to improve all of this some day and make every state very clear.

I should have written "in each poll iteration()". Poll calls account.update() in each iteration, which increments the counter, does work, then decrements the counter. Everytime the counter hits 0, it means that data is in a consistent state, and the frontend gets the newest balance, transactions, and so on.

The issue before this PR was that before poll() is even called at all, the counter is also at 0, so Balance() would return 0. 

This commit fixes it by incrementing the counter in Initialize(), and decrements it after the first account update. Bitcoin works the same way: incremented in Initialize(), and decremented to 0 once all addresses and transactions are synced.

Reply:
Yeah, no I get exactly why this change. I saw both Balance and Transactions being blocked by `WaitSynchronized` which I then traced to synchronizer's counter and a wait channel.

I guess I was complaining about me being confused :)